File: AirlinesNew/src/main/java/com/acn/service/UserService.java
package com.lti.service;

import java.util.List;

import com.lti.entity.User;

public interface UserService {

	long addAUser(User user);
	User findUserById(long userId);
	List<User> viewAllUsers();
	boolean isValidUser(String userEmail,String userPassword);
//	User isValidUser(String userEmail,String userPassword);
	String findUserNameByEmailId(String userEmail);
    boolean resetPassword(String userEmail,String userPassword);
    public long findUserIdByEmailId(String userEmail);
    public boolean checkEmail(String userEmail);
}

File: Airlinesnew/Airlines/src/app/user.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {User} from './user';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
}) 
export class UserService {
  user:User
  constructor(private http:HttpClient) {}
  public registerUser(user:User):Observable<User>{
    return this.http.post<User>('http://localhost:8989/addUser',user);
  }
  public viewAllUsers():Observable<User[]>{
    return this.http.get<User[]>('http://localhost:8989/viewAllUsers');
  }
  public findUser(userId: number): Observable<User> {
    return this.http.get<User>('http://localhost:8989/findUserById/'+userId);
  }
  public isValidUser(userEmail:string,userPassword:string){
   return this.http.get<User>('http://localhost:8989/validUser/'+userEmail+'/'+userPassword);
 
}
}